assembly/src/ast/code_body.rs

```diff
@@ -163,3 +163,34 @@ impl PartialEq for CodeBody {
         nodes && (locations || left_empty || right_empty)
     }
 }
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for CodeBody {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u64(self.nodes.len() as u64);
+        self.nodes.write_into(target);
+
+        target.write_u64(self.locations.len() as u64);
+        self.locations.write_into(target);
```

Comment:
I think we can use `u32` to encode the number of nodes. To enforce this, we can probably put an `assert!` in the constructor of `CodeBody`.

Reply:
changed